--- xv6-clean/defs.h	2026-08-24 14:26:16.379799800 +0000
+++ lab6/q2-prodcons/defs.h	2026-09-01 13:40:14.489788000 +0000
@@ -108,6 +108,10 @@
 int             fork(void);
 int             growproc(int);
 int             kill(int);
+int             shm_get(void);
+int             sem_init(int, int);
+int             sem_wait(int);
+int             sem_signal(int);
 struct cpu*     mycpu(void);
 struct proc*    myproc();
 void            pinit(void);
@@ -188,3 +192,4 @@
 
 // number of elements in fixed-size array
 #define NELEM(x) (sizeof(x)/sizeof((x)[0]))
+int             shm_map(pde_t*, void*, uint);
